lib/commands/web.js

```diff
@@ -269,12 +273,59 @@ extensions.translateWebCoords = async function (coords) {
 };
 
 extensions.checkForAlert = async function () {
+  try {
+    const alertText = await this.proxyCommand('/alert/text', 'GET');
+    if (util.hasValue(alertText)) {
+      log.debug(`Found an alert with text '${alertText}'`);
+      return true;
+    }
+  } catch (ign) {}
   return false;
 };
 
 extensions.waitForAtom = async function (promise) {
-  // TODO: Add check for alert and accept/dismiss it as per autoAcceptAlert capability
-  return this.parseExecuteResponse(await promise);
+  this._activeAtoms = this._activeAtoms || [];
+  const alertsHandler = async () => {
+    while (!_.isEmpty(this._activeAtoms)) {
+      await B.delay(ALERT_CHECK_INTERVAL);
+      if (_.isEmpty(this._activeAtoms)) {
+        return;
+      }
+
+      if (await this.checkForAlert()) {
+        if (this.opts.autoAcceptAlerts) {
+          log.debug('Accepting the alert as requested by autoAcceptAlerts capability');
+          await this.proxyCommand('/alert/accept', 'POST', {});
+        } else if (this.opts.autoDismissAlerts) {
+          log.debug('Dismissing the alert as requested by autoDismissAlerts capability');
+          await this.proxyCommand('/alert/dismiss', 'POST', {});
+        } else {
+          log.debug('Doing nothing. Set either autoAcceptAlerts or autoDismissAlerts capabilities ' +
+            'if you want alerts to be handled automatically in WEB context');
+        }
+        return;
+      }
+    }
+  };
+
+  if (_.isEmpty(this._activeAtoms)) {
+    this._activeAtoms.push(1);
+    alertsHandler().catch((e) => log.warn(e));
+  } else {
+    this._activeAtoms.push(1);
+  }
+  try {
+    return this.parseExecuteResponse(await B.resolve(promise)
+      .timeout(RESPONSE_TIMEOUT)
```

Comment:
I'm not quite sure about this timeout value. Would it be better to get it from caps or there is already some entity, which we can reuse? And how about async calls?